File: game/forest/hurt_flying_falling.py
import pygame.mixer

from animation import Animation
from audio_helper import AudioHelper
from forest.forest_resources import ForestResources
from forest.hurt_flying_falling_hang_animation import HurtFlyingFallingHangAnimation
from state import State


class HurtFlyingFalling(State):
    def process_events(self, events):
        if self.get_frame_index_fast() >= len(ForestResources.catapult_fall):
            return HurtFlyingFallingHangAnimation
        return None

    def on_enter(self) -> None:
        super().on_enter()
        AudioHelper.play(ForestResources.speak_catapult_down, self.context.audio_port)
        AudioHelper.play(ForestResources.sfx_hugo_crash, self.context.audio_port)

    def render(self, screen):
        screen.blit(Animation.get_frame(ForestResources.catapult_fall, self.get_frame_index_fast()), (0,0))

File: game-web/src/forest/hurtFlyingFalling.ts
/** Port of game/forest/hurt_flying_falling.py */

import { State, type StateClass } from '../state';
import { Animation } from '../animation';
import { AudioHelper } from '../audioHelper';
import { ForestResources } from './forestResources';
import { HurtFlyingFallingHangAnimation } from './hurtFlyingFallingHangAnimation';
import type { PhoneEvents } from '../phoneEvents';

export class HurtFlyingFalling extends State {
  processEvents(_events: PhoneEvents): StateClass | null {
    return this.getFrameIndexFast() >= ForestResources.catapultFall.length ? HurtFlyingFallingHangAnimation : null;
  }

  onEnter(): void {
    super.onEnter();
    AudioHelper.play(ForestResources.speakCatapultDown);
    AudioHelper.play(ForestResources.sfxHugoCrash);
  }

  render(ctx: OffscreenCanvasRenderingContext2D): void {
    ctx.drawImage(Animation.getFrame(ForestResources.catapultFall, this.getFrameIndexFast()), 0, 0);
  }
}
